src/client/clientErrorHandler.ts

```diff
@@ -23,11 +32,22 @@ export class ClientErrorHandler implements ErrorHandler {
 
   closed(): CloseAction {
     this.restarts.push(Date.now());
+    const heapProfileGlob = new glob.GlobSync(`${this.heapDumpFolder}/java_*.hprof`);
+    if (heapProfileGlob.found.length) {
+      // Only clean heap dumps that are generated in the default location.
+      // The default location is the extension global storage
+      // This means that if users change the folder where the heap dumps are placed,
+      // then they will be able to read the heap dumps,
+      // since they aren't immediately deleted.
+      cleanUpHeapDumps(this.context);
+      showOOMMessage();
+      return CloseAction.DoNotRestart;
+    }
     if (this.restarts.length < 5) {
       return CloseAction.Restart;
     } else {
       const diff = this.restarts[this.restarts.length - 1] - this.restarts[0];
-      if (diff <= 3 * 60 * 1000) {
+      if (diff <= 5 * 60 * 1000) {
```

Comment:
The error message just below as well. But actually, do you have an opposition to making it 10 minutes ? Basically, 1 restart every 2 minutes or less (repeated 5 times) would trigger it.

Reply:
Okay. I have set it to 5 crashes in the last 10 minutes